src/nspawn/nspawn-register.c

```diff
@@ -132,7 +132,11 @@ int register_machine(
                                            * container to ever create
                                            * these device nodes. */
                                           "/dev/pts/ptmx", "rw",
-                                          "char-pts", "rw");
+                                          "char-pts", "rw",
+                                          /* Allow /run/systemd/inaccessible/{chr,blk}
+                                           * devices inside the container */
+                                          "/run/systemd/inaccessible/chr", "rwm",
+                                          "/run/systemd/inaccessible/blk", "rwm");
```

Comment:
@alepuccetti I would suggest to allow the special strings "block-0" and "char-0". For those special strings, whitelist_major() would not read /proc/devices but just consider the major/minor zero.